External Fingerprint Storage phase 2 blog post (#3541)
jenkins-infra/jenkins.io: content/blog/2020/07/2020-07-24-external-fingerprint-storage-phase-2.adoc

This was done because it may be the case that keeping redundant fingerprints in memory might be cheaper than the
cleanup operation (especially in the case of external storages, which are cheaper these days).

=== Fingerprint Migration

* link:https://github.com/jenkinsci/jenkins/pull/4825[]

Earlier, there was no support for fingerprints stored in the local storage.
In this phase, we introduce migration support for users.
The old fingerprints are now migrated to the new configured external storage whenever they are used (lazy migration).
This allows gradual migration of old fingerprints from local disk storage to the new external storage.

=== Refactor `FingerprintStorage` to use descriptors

* link:https://github.com/jenkinsci/jenkins/pull/4834[]

* link:https://github.com/jenkinsci/redis-fingerprint-storage-plugin/pull/36[]

Earlier, whenever an external fingerprint storage plugin was installed, it was enabled by default.
We refactored the implementation to make use of `Descriptor` pattern so the fingerprint engine can now be selected
as a dropdown from the Jenkins configuration page.

> The dropdown is shown only when more than one fingerprint storage engines are configured on the system.

Reviewer: ```suggestion
The dropdown is shown only when multiple fingerprint storage engines are configured on the system.
```
